diff --git a/meson.build b/meson.build
index d530b96..caa7723 100644
--- a/meson.build
+++ b/meson.build
@@ -28,34 +28,13 @@ if sleef_dep.found() and sleef_dep.type_name() != 'internal'
   tlfloat_lib = cpp.find_library('tlfloat', required: false)
   
   if sleefquad_lib.found() and tlfloat_lib.found()
-    sleefquad_test_code = '''
-    #include <sleefquad.h>
-
-    int main(void) {
-      Sleef_quad q1 = Sleef_cast_from_doubleq1(1.0);
-      Sleef_quad q2 = Sleef_cast_from_doubleq1(2.0);
-      Sleef_quad result = Sleef_addq1_u05(q1, q2);
-      return 0;
-    }
-    '''
-    # this should compile and link
-    quad_works = cpp.links(
-      sleefquad_test_code,
-      dependencies: [sleef_dep, sleefquad_lib, tlfloat_lib],
-      name: 'SLEEF quad-precision support'
+    sleefquad_dep = declare_dependency(
+      dependencies: [sleef_dep, sleefquad_lib]
     )
-    
-    if quad_works
-      sleefquad_dep = declare_dependency(
-        dependencies: [sleef_dep, sleefquad_lib]
-      )
-      tlfloat_dep = declare_dependency(
-        dependencies: [tlfloat_lib]
-      )
-      use_system_sleef = true
-    else
-      fallback_reason = 'System-wide SLEEF installation found but a test for quad precision support failed.'
-    endif
+    tlfloat_dep = declare_dependency(
+      dependencies: [tlfloat_lib]
+    )
+    use_system_sleef = true
   else
     fallback_reason = 'System-wide SLEEF installation does not have a sleefquad library.'
   endif
